Quirks: ceac.state.gov requires document.activeElement to be HTMLInputElement

karlcow opened the pull request:
#### 0b543cc9ff31a50857a570a65f9d33f405b30a15
<pre>
Quirks: ceac.state.gov requires document.activeElement to be HTMLInputElement
<a href="https://bugs.webkit.org/show_bug.cgi?id=267569">https://bugs.webkit.org/show_bug.cgi?id=267569</a>
<a href="https://rdar.apple.com/110953808">rdar://110953808</a>

Reviewed by Aditya Keerthi.

Fixes a regression introduced by adding a new method in Quirks.
<a href="https://bugs.webkit.org/show_bug.cgi?id=260938">https://bugs.webkit.org/show_bug.cgi?id=260938</a>
There is a followup bug to improve the new method and make sure
that every type of domain matching is covered.
See <a href="https://bugs.webkit.org/show_bug.cgi?id=267623">https://bugs.webkit.org/show_bug.cgi?id=267623</a>

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsFormControlToBeMouseFocusable const):

Canonical link: <a href="https://commits.webkit.org/273282@main">https://commits.webkit.org/273282@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9a72e4bb81cf4fad7834aed362ecd214b3cdee94

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34764 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/13585 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36772 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/37454 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31373 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/35906 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15989 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10670 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30321 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35291 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11502 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30956 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10054 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10116 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31032 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38715 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31561 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31355 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36164 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10232 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8134 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/34140 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12059 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8014 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10779 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11123 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -134,7 +134,7 @@ bool Quirks::needsFormControlToBeMouseFocusable() const
         return false;
 
-    return isDomain("ceac.state.gov"_s);
+    return equalLettersIgnoringASCIICase(m_document->topDocument().url().host(), "ceac.state.gov"_s);
```
This can be a case-sensitive comparison.

karlcow replied:
@annevk do you want me to do something about this? Or do you want to merge your PR instead?

karlcow replied:
ok fixed. I didn't realize this had not been merged yet.

karlcow committed:
Quirks: ceac.state.gov requires document.activeElement to be HTMLInputElement
https://bugs.webkit.org/show_bug.cgi?id=267569
rdar://110953808

Reviewed by Aditya Keerthi.

Fixes a regression introduced by adding a new method in Quirks.
https://bugs.webkit.org/show_bug.cgi?id=260938
There is a followup bug to improve the new method and make sure
that every type of domain matching is covered.
See https://bugs.webkit.org/show_bug.cgi?id=267623

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsFormControlToBeMouseFocusable const):

Canonical link: https://commits.webkit.org/273282@main
